plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogAccessOperations.java

```diff
@@ -458,39 +442,17 @@ private void assertGlueMetastoreApiInvocations(@Language("SQL") String query, Mu
 
     private void assertGlueMetastoreApiInvocations(Session session, @Language("SQL") String query, Multiset<?> expectedInvocations)
     {
-        Map<GlueMetastoreMethod, Integer> countsBefore = Arrays.stream(GlueMetastoreMethod.values())
-                .collect(toImmutableMap(Function.identity(), method -> method.getInvocationCount(glueStats)));
-
-        getQueryRunner().execute(session, query);
-        Map<GlueMetastoreMethod, Integer> countsAfter = Arrays.stream(GlueMetastoreMethod.values())
-                .collect(toImmutableMap(Function.identity(), method -> method.getInvocationCount(glueStats)));
-
-        Map<GlueMetastoreMethod, Integer> deltas = Arrays.stream(GlueMetastoreMethod.values())
-                .collect(Collectors.toMap(Function.identity(), method -> countsAfter.get(method) - countsBefore.get(method)));
-        ImmutableMultiset.Builder<GlueMetastoreMethod> builder = ImmutableMultiset.builder();
-        deltas.entrySet().stream().filter(entry -> entry.getValue() > 0).forEach(entry -> builder.setCount(entry.getKey(), entry.getValue()));
-        Multiset<GlueMetastoreMethod> actualInvocations = builder.build();
-
-        if (expectedInvocations.equals(actualInvocations)) {
-            return;
-        }
-
-        List<String> mismatchReport = Sets.union(expectedInvocations.elementSet(), actualInvocations.elementSet()).stream()
-                .filter(key -> expectedInvocations.count(key) != actualInvocations.count(key))
-                .flatMap(key -> {
-                    int expectedCount = expectedInvocations.count(key);
-                    int actualCount = actualInvocations.count(key);
-                    if (actualCount < expectedCount) {
-                        return Stream.of(format("%s more occurrences of %s", expectedCount - actualCount, key));
-                    }
-                    if (actualCount > expectedCount) {
-                        return Stream.of(format("%s fewer occurrences of %s", actualCount - expectedCount, key));
-                    }
-                    return Stream.of();
-                })
-                .collect(toImmutableList());
-
-        fail("Expected: \n\t\t" + join(",\n\t\t", mismatchReport));
+        DistributedQueryRunner queryRunner = getDistributedQueryRunner();
+        spanExporter.reset();
```

Comment:
this shouldn't be needed. you could filter captures span id's by the query id which is added as the attribute

Reply:
I don't think we can do that when injecting `OpenTelemetry` object directly.

Reply:
Yes. But spans have trino_query_id attribute attached

Reply:
Nope. 
<img width="674" alt="Screenshot 2023-07-21 at 19 03 37" src="https://github.com/trinodb/trino/assets/6237050/fb72e30b-1ff5-4fcd-b858-450ffd7ad90a">